protocols/cosi/cosi.go

```diff
@@ -289,39 +266,22 @@ func (pc *ProtocolCosi) sendChallenge(out *CosiChallenge) error {
 
 }
 
-func (pc *ProtocolCosi) StartResponse() error {
-	// TODO check the hook
-	// else do it yourself
-	resp, err := pc.Cosi.CreateResponse()
-	if err != nil {
-		return err
-	}
-	out := &CosiResponse{
-		Response: resp,
-	}
-	dbg.Lvl3(pc.Node.Name(), "ProtocolCosi().StartResponse()")
-	err = pc.SendTo(pc.Parent(), out)
-	pc.Cleanup()
-	return err
-}
-
 // handleResponse brings up the response of each node in the tree to the root.
 func (pc *ProtocolCosi) handleResponse(in *CosiResponse) error {
-	// add to temporary
-	pc.tempResponseLock.Lock()
-	pc.tempResponse = append(pc.tempResponse, in)
-	pc.tempResponseLock.Unlock()
-	// do we have enough ?
-	dbg.Lvl3(pc.Node.Name(), "has", len(pc.tempResponse), "responses")
-	if len(pc.tempResponse) < len(pc.Children()) {
-		return nil
+	if !pc.IsLeaf() {
+		// add to temporary
+		pc.tempResponseLock.Lock()
+		pc.tempResponse = append(pc.tempResponse, in)
```

Comment:
Yes, I understood that. I was just wondering if it isn't obsolete to add the `nil` to the underlying `pending`... as it comes with costly  `Lock`/`Unlock` (nah, not really that costly, but if obsolete it can be removed)